x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_preview/api/preview_rules/route.ts

```diff
@@ -197,45 +198,11 @@ export const previewRulesRoute = (
             isPreview: true,
           });
 
-          const runExecutors = async <
-            TParams extends RuleParams,
-            TState extends RuleTypeState,
-            TInstanceState extends AlertInstanceState,
-            TInstanceContext extends AlertInstanceContext,
-            TActionGroupIds extends string = ''
-          >(
-            executor: ExecutorType<
-              TParams,
-              TState,
-              TInstanceState,
-              TInstanceContext,
-              TActionGroupIds
-            >,
-            ruleTypeId: string,
-            ruleTypeName: string,
-            params: TParams,
-            shouldWriteAlerts: () => boolean,
-            alertFactory: {
-              create: (
-                id: string
-              ) => Pick<
-                Alert<TInstanceState, TInstanceContext, TActionGroupIds>,
-                | 'getState'
-                | 'replaceState'
-                | 'scheduleActions'
-                | 'setContext'
-                | 'getContext'
-                | 'hasContext'
-                | 'getUuid'
-                | 'getStart'
-              >;
-              alertLimit: {
-                getValue: () => number;
-                setLimitReached: () => void;
-              };
-              done: () => { getRecoveredAlerts: () => [] };
-            }
+          const runExecutors = async <TParams extends RuleParams, TState extends RuleTypeState>(
+            securityRuleType: SecurityAlertType<TParams, TState>,
+            params: TParams
           ) => {
+            const ruleType = previewRuleTypeWrapper(securityRuleType);
```

Comment:
Nit: I always have to look up what these functions are doing, because it's not self-evident from the name and there is no documentation for them. This is more of an adapter, though, right?

```suggestion
            const frameworkRuleType = securityRuleTypeAdapter(securityRuleType);
```

Reply:
But: isolating its usage to a single call _behind_ this API is a huge improvement either way 👍